source/OptimaJet.DWKit.StarterApplication/Forms/Products/BaseProductForm.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Linq;
+using Optimajet.DWKit.StarterApplication.Models;
+using OptimaJet.DWKit.StarterApplication.Repositories;
+using OptimaJet.DWKit.StarterApplication.Services;
+
+namespace OptimaJet.DWKit.StarterApplication.Forms.Products
+{
+    public class BaseProductForm : BaseForm
+    {
+        public BaseProductForm(
+            UserRepository userRepository,
+            ICurrentUserContext currentUserContext) : base(userRepository, currentUserContext)
+        {
+        }
+        protected void ValidateProduct(Project project, ProductDefinition productDefinition)
+        {
+            if ((project == null) || (productDefinition == null))
+            {
+                // Allowing it to return in these cases should cause the base to hit normal
+                // foreign key failures
+                return;
+            }
+            if ((project.Organization == null))
+            {
+                var message = $"The project for this product does not have a valid organization";
+                AddError(message);
+                return; // Can't proceed further without causing exceptions
+            }
+
+            if (!project.Organization.ProductDefinitionIds.Contains(productDefinition.Id))
+            {
+                var message = $"This product is not permitted for the organization";
+                AddError(message);
+            }
+            // The current user should be a member of the organization
+            if (!CurrentUserOrgIds.Contains(project.Organization.Id))
+            {
+                var message = ("The current user is not a member of the product organization");
```

Comment:
message should say project?

Reply:
From what Chris was after me about in the other message, I guess this one should just say that the current user is not a member of the organization. The reasons for removing project in the other message was "It just happens to be connected to a project.  It wouldn’t be allowed for any project of the organization so “project” is just a container of products and not relevant to the check. The project is inherently tied to a organization.  It is a container.  It really isn’t relevant to the check other than the way to navigate to the organization.
You are really checking, “Does this organization allow this type of product”.  I would assume that the same rationale would apply here and change the message to "The current user is not a member of the organization"